src/unified_recon/rule_0/decomposer.py

```
@@ -0,0 +1,106 @@
+        Args:
+            decomposition_patterns: Dict with "crack" and/or "spread" patterns
+        """
+        self.patterns = decomposition_patterns or {}
+        self.crack_pattern = self.patterns.get("crack")  # e.g., "brent swap" for ICE
+        self.spread_pattern = self.patterns.get("spread")  # e.g., "-" for spreads
+    
+    def decompose(
+        self, 
+        product_name: str, 
+        quantity: Decimal,
+        buy_sell: str
```

Ruff (0.12.2)

Unused method argument: `buy_sell`

(ARG002)